# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/ffmpeg/FFmpegVideoDecoder.h
# Commit: c159181ff5d0
# Full Hash: c159181ff5d00814ffa6c1648dc22002b70da9ed
# Author: Martin Stransky <stransky@redhat.com>
# Date: 2020-04-21 22:25:03
# Regressor Bug: 1630754
# File Overlap Count: 1
# Description:
#   Bug 1630754 [Wayland][VA-API] Explicitly configure VADisplay display for VA-API video playback, r=jya
#   
#   Some gfx drivers (especially on a child process) can open X11 VADisplay instead of a Wayland one which leads
#   to HW playback failure. As a solution let's create VADisplay explicitly on top of our wayland display connection.
#   
# ==============================================================================

diff -r 27d3c6f3204e -r c159181ff5d0 dom/media/platforms/ffmpeg/FFmpegVideoDecoder.h
--- a/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.h	Tue Apr 21 10:50:17 2020 +0000
+++ b/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.h	Tue Apr 21 10:52:47 2020 +0000
@@ -111,6 +111,7 @@
 #ifdef MOZ_WAYLAND_USE_VAAPI
   AVBufferRef* mVAAPIDeviceContext;
   const bool mDisableHardwareDecoding;
+  VADisplay mDisplay;
 #endif
   RefPtr<KnowsCompositor> mImageAllocator;
   RefPtr<ImageContainer> mImageContainer;